pages/agent/v3/signed-pipelines.md

```diff
@@ -0,0 +1,114 @@
+```bash
+buildkite-agent tool keygen --alg PS256 --key-id my-key-id
+```
+
+The agent will then generate two JSON Web Key Sets for you, one public and one private, and output them to files in the current directory. You can then use these keys to sign and verify your pipelines.
+
+If no key id is provided, the agent will generate a random one for you.
+
+Note that the value of `--alg` must be a valid [JSON Web Signing Algorithm](https://datatracker.ietf.org/doc/html/rfc7518#section-3). The agent does not support the `none` algorithm, or any signature algorithms in the `RS` series (any signing algorithms that use RSASSA-PKCS1 v1.5 signatures, `RS256` for example).
+
+<details>
+  <summary>Why doesn't the agent support RSASSA-PKCS1 v1.5 signatures?</summary>
+  In short, it's because RSASSA-PKCS1 v1.5 signatures are generally regarded to be less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still largely known to be relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures, which are generally regarded to be more secure than RSA signatures.
+</details>
+
+### What algorithm should I use?
+
+Signed Pipelines supports a number of different signing algorithms, which can largely be broken down into two categories: Symmetric Signing Algorithms (where signing and verification use the same keys), and Asymmetric Signing Algorithms (where signing and verification use different keys).
+
+We recommend using an asymmetric signing algorithm, as it creates a permissions boundary between the agents that upload and sign pipelines, and agents that run other jobs and have the ability to verify signatures. This means that were an attacker able to compromise instances with verification keys, they wouldn't be able to modify the steps that are run on your agents.
+
+With regards to specific algorithm choice, basically any of the asymmetric signing algorithms we support are fine and will be secure. If you're not sure which one to use, `EdDSA` is proven to be secure, has a modern design, wasn't designed by a Nation State Actor, and produces nice short signatures.
+
+Signed pipelines supports the following signing algorithms:
+
+#### Symmetric
+- `HS256`
+- `HS384`
+- `HS512`
+
+#### Asymmetric
+- `PS256`
+- `PS384`
+- `PS512`
+- `ES256`
+- `ES384`
+- `ES512`
+- `EdDSA`
+
+### Making your agents sign steps they upload 
+
+Now that your keys are generated, we need to configure your agents to use them. On agents that will be uploading pipelines, add the following to your agent's config file:
+```ini
+job-signing-jwks-path=<path to signing keys>
+signing-key-id=<the key id you generated earlier>
+job-verification-jwks-path=<path to verification keys>
```

Comment:
Does the signing agent require the verification key? If so, I'd note here why - "the signing agent requires the verification key as well in order to validate the signature" or whatever the reason is.

Reply:
ahah, i was in the process of writing that section when i committed this. tl;dr, if we statically sign the initial steps, the uploader needs to both be able to verify the signature on the initial steps, and sign the steps it uploads from the repo

Reply:
Ah of course!